server/proxy/modules/dyn-channel-dump/dyn-channel-dump.cpp

```diff
@@ -0,0 +1,332 @@
+	                           data->name) != plugin_static_intercept.end();
+	if (intercept)
+		data->intercept = TRUE;
+	return TRUE;
+}
+
+static BOOL dump_dyn_channel_intercept(proxyPlugin* plugin, proxyData* pdata, void* arg)
+{
+	auto data = static_cast<proxyDynChannelInterceptData*>(arg);
+
+	WINPR_ASSERT(plugin);
+	WINPR_ASSERT(pdata);
+	WINPR_ASSERT(data);
+
+	data->result = PF_CHANNEL_RESULT_PASS;
+	if (dump_channel_enabled(plugin, pdata, data->name))
+	{
+		auto cdata = dump_get_plugin_data(plugin, pdata);
+		auto& stream = cdata->stream(data->name, data->isBackData);
+		auto buffer = reinterpret_cast<const char*>(Stream_ConstBuffer(data->data));
+		if (!stream.is_open() || !stream.good())
+		{
+			WLog_ERR(TAG, "Could not write to stream");
+			return FALSE;
+		}
+		stream.write(buffer, Stream_Length(data->data));
+		if (stream.fail())
+		{
+			WLog_ERR(TAG, "Could not write to stream");
+			return FALSE;
+		}
+	}
+
+	return TRUE;
+}
+
+static std::vector<std::string> split(const std::string& input, const std::string& regex)
+{
+	// passing -1 as the submatch index parameter performs splitting
+	std::regex re(regex);
+	std::sregex_token_iterator first{ input.begin(), input.end(), re, -1 }, last;
+	return { first, last };
+}
+
+static BOOL dump_session_started(proxyPlugin* plugin, proxyData* pdata, void*)
+{
+	WINPR_ASSERT(plugin);
+	WINPR_ASSERT(pdata);
+
+	auto config = pdata->config;
+	WINPR_ASSERT(config);
+
+	auto cpath = pf_config_get(config, plugin_name, key_path);
+	if (!cpath)
+	{
+		WLog_ERR(TAG, "Missing configuration entry [%s/%s], can not continue", plugin_name,
+		         key_path);
+		return FALSE;
+	}
+	auto cchannels = pf_config_get(config, plugin_name, key_channels);
+	if (!cchannels)
+	{
+		WLog_ERR(TAG, "Missing configuration entry [%s/%s], can not continue", plugin_name,
+		         key_channels);
+		return FALSE;
+	}
+
+	std::string path(cpath);
+	std::string channels(cchannels);
+	std::vector<std::string> list = split(channels, "[;,]");
+
+	if (!std::filesystem::exists(path))
+	{
+		if (!std::filesystem::create_directories(path))
+		{
+			WLog_ERR(TAG, "Failed to create dump directory %s", path.c_str());
+			return FALSE;
+		}
+	}
+	else if (!std::filesystem::is_directory(path))
+	{
+		WLog_ERR(TAG, "dump path %s is not a directory", path.c_str());
+		return FALSE;
+	}
+
+	if (list.empty())
+	{
+		WLog_ERR(TAG, "Empty configuration entry [%s/%s], can not continue", plugin_name,
+		         key_channels);
+		return FALSE;
+	}
+	dump_set_plugin_data(plugin, pdata, new ChannelData(path, list));
+
+	return TRUE;
+}
+
+static BOOL dump_session_end(proxyPlugin* plugin, proxyData* pdata, void*)
+{
+	WINPR_ASSERT(plugin);
+	WINPR_ASSERT(pdata);
+
+	dump_set_plugin_data(plugin, pdata, nullptr);
+	return TRUE;
+}
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+	FREERDP_API BOOL proxy_module_entry_point(proxyPluginsManager* plugins_manager, void* userdata);
```

Comment:
Since we obviously use cpp here we can just write `extern "C" FREERDP_API BOOL proxy_module_entry_point ...`

Reply:
good point